Sending attributes from <auth>/send section.

HelenMamontova opened the pull request:
Class RAD_SETTINGS is transferred to separate files .h / .cpp. The makeResponse function changed to send attributes configured in the <auth>/send section. Function makeAttributes added.

HelenMamontova committed:
Radius. Header file <mutex>, <jthread.hpp> added. Functions start, stop
declaration changed. Class members m_thread, m_mutex added. Function Run
declaration added.

HelenMamontova committed:
Radius. Header files added. Functions Start, Stop, Run definition added.

HelenMamontova committed:
Radius. New files server.h, server.cpp added.

HelenMamontova committed:
Radius. Definition of functions startReceive, makeResponse, handleSend,
handleReceive added.

HelenMamontova committed:
The Run function name fixed. Definition of secret variable sdded.

HelenMamontova committed:
The file server.cpp added to add_library command, command
find_package(OpenSSL 1.0.0 Required) added, OpenSSL::Crypto added to
target_link_libraries command in the block if(BUILD_MOD_RADIUS).

HelenMamontova committed:
Parameter token added to function Run() declaration.

HelenMamontova committed:
Radius. Method Start: variable isRunning=true removed, m_thread
definition changed. Method Stop: variable isRunning=false removed,
m_thread.joinable check added, isRunning check added, request_stop call added.
Method Run: parameter token added, variables secret, port removed,
object lock and isRunning=true added before cycle while,
cycle while added, isRunning=false added after cycle while.

HelenMamontova committed:
Radius. Hold the mutex removed,extra conditions
for m_thread.join() removed in the Stop function.

HelenMamontova committed:
Radius. Cycle while removed in function Run.

HelenMamontova committed:
Radius. The variables isRunning, errorStr replaced by m_running,
m_errorStr in class Radius.

HelenMamontova committed:
Radius. The variable isRunning replaced by m_running in function Run.

HelenMamontova committed:
Radius. Object name io_service changed to ioService in the function Run.

HelenMamontova committed:
Method SetRunning declaration added to the class RADIUS.

HelenMamontova committed:
Method SetRunning definition added. Method SetRunning call added to
function Run.

HelenMamontova committed:
Radius. Namespace STG added.

HelenMamontova committed:
Radius. Declaration using STG::Server added.

HelenMamontova committed:
Radius. Header file "stg/logger.h" added. RADIUS class member m_loger added.

HelenMamontova committed:
Radius. Header file "stg/common.h" added. Initialization of m_logger
added to constructor RADIUS. Output cerr replaced by logger and
printfd().

HelenMamontova committed:
Radius. Variable except added, messages fixed in the function Run.

HelenMamontova committed:
Radius. Thread join logic fixed in the function Stop.

HelenMamontova committed:
Keyword const added to std::lock_guard in the function SetRunning.

HelenMamontova committed:
Radius. Class member m_mutex moved to the top of list.

HelenMamontova committed:
Radius. Class member m_logger put after the m_running.

HelenMamontova committed:
Radius. Unnecessary variable except removed in the function Run.

HelenMamontova committed:
Radius. Extra whitespace removed in the function Stop. Function
IsRunning definition added.

HelenMamontova committed:
Radius. Function IsRunning declaration changed.

HelenMamontova committed:
Radius. Formatting fixed.

HelenMamontova committed:
Radius. Extra symbols '//' removed.

HelenMamontova committed:
Radius. Class member m_running initialization added to constructor
RADIUS.

HelenMamontova committed:
Radius.Header file "stg/module_settings.h" added. Class members variable
m_settings and method SetSettings added to class RADIUS.

HelenMamontova committed:
Radius. The parameters Secret, Dictionaries added, parameter Port
changed, other parameters removed.

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius

HelenMamontova committed:
Radius. Header files added. Class RAD_SETTINGS added. Methods
SetSettings, copy constructor RADIUS, assignment operator added. Method
ParseSettings changed. Formatting fixed.

HelenMamontova committed:
Radius. Parameter default changed to optional and parameters value
commented out for the parameters dictionaries, port.

HelenMamontova committed:
Radius. Description of parameters secret, port, dictionaries changed.

HelenMamontova committed:
Radius. Copy constructor RADIUS and assignment operator declaration
changed and moved to public section.

HelenMamontova committed:
Radius. Class member m_mutex moved to the top of section private.

HelenMamontova committed:
Radius. The m_port variable value initialization changed to 1812,
m_dictionaries variable initialization added to constructor RAD_SETTINGS.

HelenMamontova committed:
Radius. Check for missing parameters port and dictionaries removed in function ParseSettings.

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius

HelenMamontova committed:
Radius. The dist directory added.

HelenMamontova committed:
Radius. The methods declaration stop and start added.

HelenMamontova committed:
Radius. The methods stop, start definition added. Method start call
added to Server constructor.

HelenMamontova committed:
Radius. The header files "server.h", <boost/asio.hpp>, <memory> added.
The class members m_ioService, m_server added.

HelenMamontova committed:
Radius. The header files "server", <boost/asio.hpp>, <string> removed.
The method m_server->stop() call added to method Stop.

HelenMamontova committed:
Radius. Header files <stop_token> and <jthread.hpp> added. Parameter
token added to constructor Server. Class member m_token added.

HelenMamontova committed:
Radius. Parameter token and m_token class member initialization added to Server
constructor definition. The check m_token.stop_requested() added to
method handleSend before startReceive method call.

HelenMamontova committed:
Radius. The m_server class member initialization removed.

HelenMamontova committed:
Radius. The m_server->stop() call adnd m_thread.request_stop() call
reodered, the m_server check added before m_server->stop call in the
method Stop. Parameter token added to constructor Server in the function
Run.

HelenMamontova committed:
Radius. Header file "stg/logger.h" added. Class member m_logger added to
class Server.

HelenMamontova committed:
Radius. Header file "stg/common.h" added, header file <iostream>
removed. Class member m_logger initialization added to constructor
Server. Object std::cout replaced by function printfd and m_logger.

HelenMamontova committed:
Radius. Class member m_logger initialization changed in the constructor
Server. Check token moved before checking the error code in the
handleSend method.

HelenMamontova committed:
Radius. Check token added to handleReceive method.

HelenMamontova committed:
Radius. Header file <stop_token> replaced by <stop_token.hpp>, header
file <jthread.hpp> removed.

HelenMamontova committed:
Radius. Parameter m_logger added to constructor Server call in the
method Run.

HelenMamontova committed:
Radius. Parameter logger added to constructor Server declaration. Class
member m_logger added to class Server.

HelenMamontova committed:
Radius. Parameter logger added to constructor Server definition. Class
member m_logger initialization changed in the constructor Server.

HelenMamontova committed:
Radius. Extra whitespace removed ib the method Run.

HelenMamontova committed:
Radius. The order of operations changed in the method handleSend.

HelenMamontova committed:
Radius. The order of operations changed in the method handleReceive.

HelenMamontova committed:
Radius. Operator return removed when checking ec, operator else removed
when checking packet in the handleReceive method.

HelenMamontova committed:
Radius. Parameter token replaced by std::move(token) in the constructor Server
in method Run.

HelenMamontova committed:
Radius. Class member m_token initialization value changed to
std::move(token) in the constructor Server.

HelenMamontova committed:
Radius. The c_str() function added to message when checking error in the
handleSend and handleReceive methods.

HelenMamontova committed:
Radius. Header file <cstring> added.

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius

HelenMamontova committed:
Radius. Header files "stg/users.h", "stg/user.h", "stg/user_property.h"
added. The using statements added. The findUser, SetUsers functions
declaration added. Class member m_users added.

HelenMamontova committed:
Radius. Header files "radproto/attribute_types.h", <cstdint> added.
The m_users class member initialization added to constructor Server. The
findUser function definition added. The findUser function call added to
makeResponse function.

HelenMamontova committed:
Radius. Getting the values of login and password fixed in the function
findUser. Extra functions printfd call added.

HelenMamontova committed:
Radius. Using statements added. Method SetUsers added, variable class
member m_users added to class RADIUS.

HelenMamontova committed:
Radius. The class member variable m_users initialization added to
constructor Radius. Parameter m_users added to constructor Server call
in the method Run.

HelenMamontova committed:
Radius. Parameter users added to constructir Server declaration.
Function SetUsers removed fron class Server.

HelenMamontova committed:
Radius. Parameter users added to constructor Server definition. Variable
class member m_users initialization added to constructor Server.

HelenMamontova committed:
Radius. Function c_str added to login and password in m_logger and
printfd function in the function findUser.

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius

HelenMamontova committed:
Radius. Extra functions printfd removed.

HelenMamontova committed:
Radius. The using declaration for UserPtr, ConstUserPtr removed.

HelenMamontova committed:
Radius. Looking for user moved after check request type in the
makeResponse function.

HelenMamontova committed:
Radius. The using declaration for UserPtr, ConstUserPtr removed. Header
files "stg/user.h", "stg/users.h", "stg/user.property.h" removed.

HelenMamontova committed:
Radius. Header files "stg/user.h", "stg/users.h" added.

HelenMamontova committed:
Radius. Function findUser moved to private.

HelenMamontova committed:
Radius. Return type of function findUser changed to bool.

HelenMamontova committed:
Radius. Return type of function findUser definition and call changed to bool.

HelenMamontova committed:
Radius. Variables attributesEmpty, vendorSpecificEmpty added for packet
ACCESS_REJECT in the function makeResponse.

HelenMamontova committed:
Radius. Type STG::Users* replaced by Users* in parameter users in the
constructor Server definition.

HelenMamontova committed:
Radius. Condition and code order changed when packet is returned in
makeResponse function.

HelenMamontova committed:
Radius. Unnecessary variables attributeEmpty, vendorSpecificEmpty
replaced by {} when ACCESS_REJECT packet is returned in makeResponse
function.

HelenMamontova committed:
Radius. Prefix STG of user object removed, user is initialized by
nullptr and moved to the point before call FindByName function in the
findUser function.

HelenMamontova committed:
Radius. Extra printfd function call removed in the function findUser.

HelenMamontova committed:
Radius. The way of initializing the object user changed in the findUser
function. Error log removed in the makeResponse function.

HelenMamontova committed:
Radius. Prefix STG of class member m_users and of parameter u in
SetUsers function removed.

HelenMamontova committed:
Radius. Prefix STG of class member m_users and of parameter users
in the constructor Server declaration removed.

HelenMamontova committed:
Radius. The messages of m_logger and printfd changed in function
findUser.

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius

HelenMamontova committed:
Radius. Search for parameters auth and send added to the ParseSettings function.

HelenMamontova committed:
Radius. Extra whitespace removed, unnecessary variables sendValue and
sectionsParams removed in ParseSettings function.

HelenMamontova committed:
Radius. The struct AttrValue added.

HelenMamontova committed:
Radius. Header files <boost/tokenizer.hpp>, <vector>, <utility> added.
Parsing attributes of field send in section auth of mod_radius.conf
added to the function ParseSettings.

HelenMamontova committed:
Radius. Filling vector of pairs attrSend added. Extra printfd function
removed.

HelenMamontova committed:
Radius. The Sign enumeration elements names changed in the struct
AttrValue.

HelenMamontova committed:
Radius. Using declaration for struct AttrValue added. Function ParseSendAttr definition added
and call in the function ParseSettings added.

HelenMamontova committed:
Radius. The enum class Sign renamed to Type, object sign renamed to
type.

HelenMamontova committed:
Radius. The enum class Sign renamed to Type, object sign renamed to type
in ParseSendAttr function and ParseSettings function.

HelenMamontova committed:
Radius. The enumerator names NOT_VALUE changed to PARAN_NAME, IS_VALUE
changed to VALUE in enum Type.

HelenMamontova committed:
Radius. Radius. The enumerator names NOT_VALUE changed to PARAM_NAME, IS_VALUE changed to VALUE in
ParseSendAttr function.

HelenMamontova committed:
Radius. The AttrValue struct moved into the RAD_SETTINGS class.

HelenMamontova committed:
Radius. The using statement for struct AttrValue removed. The using alias for
struct AttrValue added.

HelenMamontova committed:
Radius. The vector parsedSendAttr check added to the function
ParseSendAttr.

HelenMamontova committed:
Radius. The ParseSendAttr method added, m_logger variable class member
added to RAD_SETTINGS class.

HelenMamontova committed:
Radius. ParseSendAttr function changed to method of RAD_SETTINGS class.
Vector parsedSendAttr check for emptiness and check vector size added to
the ParseSendAttr method. Class member m_logger initialization added to
initialization list to RAD_SETTINGS constructor.

HelenMamontova committed:
Radius. Quotes check for the ValueName added to the ParseSendAttr
method.

HelenMamontova committed:
Radius. Extra variables attrValue.value and attrValue.type removed in
the ParseSendAttr method.

HelenMamontova committed:
Radius. Extra variable attrValue removed in ParseSendAttr method.

HelenMamontova committed:
Radius. Send attributes parsing changed in method ParseSendAttr.

HelenMamontova committed:
Radius. Argument fieldSendAttr name changed to value and argument type
changed to const std::string& in ParseSendAttr method. Vector
keyValuePairs name changed to res.

HelenMamontova committed:
Radius. Argument fieldSendAttr name changed to value and argument type
changed to const std::string& in ParseSendAttr method.

HelenMamontova committed:
Radius. Vector name parsedSendAttr changed to keyValue in ParseSendAttr
method.

HelenMamontova committed:
Radius. Functions front() and back() are used to get the first and last
characters of string valueName in ParseSendAttr method.

HelenMamontova committed:
Radius. Method declaration ParseSendAttr moved to private in class
RAD_SETTINGS.

HelenMamontova committed:
Radius. Formatting fixed. Class member m_res and public function
GetRes added to class RAD_SETTINGS.

HelenMamontova committed:
Radius. Extra variable res removed. Class member m_res definition added
to ParseSendAttr function and ParseSettings function.

HelenMamontova committed:
Radius. Vector res added, variable m_res replaced by res in
ParseSendAttr function.

HelenMamontova committed:
Radius. Function name GetRes changed to GetSendPairs, class member
vaiable name m_res changed to m_sendPairs in class RAD_SETTINGS.

HelenMamontova committed:
Radius. Class member vaiable name m_res changed to m_sendPairs in the
ParseSettings function.

HelenMamontova committed:
Radius. Parameter auth added to mod_radius.conf.

HelenMamontova committed:
Radius. The ParseMatchAttr method declaration and m_matchPairs class
member added to RAD_SETTINGS class.

HelenMamontova committed:
Radius. The ParseMatchAttr function definition added. Match attributes
parsing added to ParseSetting function.

HelenMamontova committed:
Radius. The ParseMatchAttr method removed, the ParseSendAttr method
renamed to ParseAuthAttr in the class RAD_SETTINGS.

HelenMamontova committed:
Radius. The ParseSendAttr method definition and call renamed to
ParseAuthAttr. Parameter paramName added to ParseAuthAttr method. The
ParseMatchAttr method definition removed. ParseMatchAttr method call
replaced by ParseAuthAttr.

HelenMamontova committed:
Radius. Attribute for parameter auth/match changed.

HelenMamontova committed:
Radius. Parameter autz added.

HelenMamontova committed:
Radius. ParseAuthAttr function renamed to ParseAuthAutzAttr.

HelenMamontova committed:
Radius. ParseAuthAttr function renamed to ParseAuthAutzAttr. The autz
attributes parsing added, m_logger and printfd messages changed in the
ParseAuthAutzAttr function.

HelenMamontova committed:
Radius. ParseAuthAutzAttr function renamed to ParseSectionsAttr. The MakeKeyValuePairs function
definition added.

HelenMamontova committed:
Radius. ParseAuthAutzAttr function renamed to ParseSectionsAttr. The messages
of m_logger and printfd changed in the ParseSectionsAttr function.
The MakeKeyValuePairs function definition and call added.

HelenMamontova committed:
Radius. Attributes of auth/send, autz/send, autz/match changed.

HelenMamontova committed:
Radius. ParseSectionsAttr function name changed to ParseRyles.

HelenMamontova committed:
Radius. ParseSectionsAttr function name definition and call changed to ParseRyles.

HelenMamontova committed:
Radius. Extra printfd function call removed in MakeKeyValuePairs
functions.

HelenMamontova committed:
Radius. The ShowRules, GetSendPairsAuth, GetMatchPairsAuth, GetSendPairsAutz, GetMatchPairsAutz
functions declaration added. The m_sendPairs, m_matchPairs class members
removed. The m_sendPairsAuth, m_matchPairsAuth, m_sendPairsAutz, m_matchPairsAutz
class members added.

HelenMamontova committed:
Radius. The ShowRules function definition added and its call in
MakeKeyValuePairs function added.

HelenMamontova committed:
Radius. The strRules string creating fixed and strRules string renamed to result
in the ShowRules function.

HelenMamontova committed:
Radius. Parameter name changed to attributes and & added to parameter in
the ParseRules function.

HelenMamontova committed:
Radius. The ShowRules function definition changed. Parameter name changed to attributes and & added to parameter in the ShowRules function.

HelenMamontova committed:
Radius. The struct ASection added. The getAuth, getAutz, parseASection
function declaration added. Class member m_auth, m_autz added.
Unnecessary code removed.

HelenMamontova committed:
Radius. The using ASection for alias added. The parseASection function
definition and call added, MakeKeyValuePairs function removed,
ParseSettings function changed. Extra printfd function added.

HelenMamontova committed:
Radius. The return added to parseASection function. Extra printfd
functions removed.

HelenMamontova committed:
Radius. GIT_TAG changed to 1.1.3 in the ExternalProject_Add for
async-radius.

HelenMamontova committed:
Radius. The m_ioService class member replaced by m_ioContext in class
Radius.

HelenMamontova committed:
Radius. The m_ioService class member replaced by m_ioContext in the
function Run.

HelenMamontova committed:
Radius. Parameter io_service replaced by io_context in the constructor Server.

HelenMamontova committed:
Radius. Parameter io_service replaced by io_context in the constructor Server definition.

HelenMamontova committed:
Radius. Parameters msg, ip commented out in the SendMessage function.

HelenMamontova committed:
Radius. Variable 'sep' declared 'const' in ParseRules function.

HelenMamontova committed:
Radius. Variables 'tokens', 'sp', 'tok' declared 'const' in ParseRules function.

HelenMamontova committed:
Radius. ShowRules function declaration removed from class RAD_SETTINGS.

HelenMamontova committed:
Radius. ShowRules function definition is made as a free function.

HelenMamontova committed:
Radius. ShowRules function definition is placed in the anonymous namespace.

HelenMamontova committed:
Radius. The code for determining the first element of vector changed in the function ShowRules.

HelenMamontova committed:
Radius. Function SetUsers declaration is made as override in the class
Radius.

HelenMamontova committed:
Radius. Header file <algorithm> added.

HelenMamontova committed:
Radius. The rad_settings.cpp file name added to add_library for MODULE
radius.

HelenMamontova committed:
Radius. Header file "rad_settings.h" added. RAD_SETTINGS class
declaration removed, struct Settings forward declaration removed.

HelenMamontova committed:
Radius. Header files <boost/tokenizer.hpp>, <vector>, <iterator>,
    <algorithm>, <ioservice> removed. Extra using declaration removed.
    Functions definition of class RAD_SETTINGS removed. Parameter
    m_radSettings added to Server constructor call.

HelenMamontova committed:
Radius. The new file rad_settings.h added.

HelenMamontova committed:
Radius. The new file rad_settings.cpp added.

HelenMamontova committed:
Radius. Header file "rad_settings.h" added. Parameter radSettings added to
constructor Server declaration. Class memberm_radSettings added.

HelenMamontova committed:
Radius. Parameter radSettings added to constructor Server definition. Class member m_radSettings initialization added in the constructor Server.

HelenMamontova committed:
Radius. The attributes User-Name, Cacllback-Number added to section <auth>/send.

HelenMamontova committed:
Radius. Class User forward declaration added. Parameter radSettings
changed to const in constructor Server declaration. The bool findUser function
declaration replaced by const User* findUser.

HelenMamontova committed:
Radius. Header files added. The using statement for STG User added. Parameter radSettings
changed to const in constructor Server definition. The bool findUser function definition
replaced by const User* findUser. The makeAttributes function definition
added. Function makeResponse definition changed.

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius
Conflicts resolved.

madf commented on projects/stargazer/inst/linux/etc/stargazer/conf-available.d/mod_radius.conf:
```diff
@@ -18,7 +18,7 @@
 
     <auth>
-        send = "Framed-IP-Address = currip, Service-Type = \'Framed-User\'"
+        send = "User-Name = login, Framed-IP-Address = currip, Service-Type = \'Framed-User\', Callback-Number = \'987654abc\'"
```
Let's keep the original value. It makes more sense.

madf commented on projects/stargazer/plugins/CMakeLists.txt:
```diff
@@ -15,7 +15,8 @@ if ( BUILD_MOD_RADIUS )
     add_library ( mod_radius MODULE other/radius/radius.cpp
-                                    other/radius/server.cpp)
+                                    other/radius/server.cpp
+                                    other/radius/rad_settings.cpp)
```
How about just `settings.cpp`?

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -4,17 +4,23 @@
     : m_radius(io_context, secret, port),
       m_dictionaries(filePath),
       m_users(users),
+      m_radSettings(radSettings),
```
Let's call them `settings` and `m_settings`.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -37,27 +43,78 @@ void Server::startReceive()
 }
 
-RadProto::Packet Server::makeResponse(const RadProto::Packet& request)
+std::vector<RadProto::Attribute*> Server::makeAttributes(const User* user)
```
I think this code is derived from the `async-radius` library, isn't it? If so, let's just expose it in the lib and use here.

HelenMamontova committed:
Radius. Unnecessary attributes removed from <auth>/send section.

HelenMamontova committed:
Radius. The rad_settings.h header file changed to settings.h.

HelenMamontova committed:
Radius. The rad_settings.h header file changed to settings.h.

HelenMamontova committed:
Radius. The rad_settings.h header file changed to settings.h.

HelenMamontova committed:
Radius. The rad_settings.cpp filename changed to settings.cpp in
add_library for BUILD_MOD_RADIUS.

HelenMamontova committed:
Radius. The settings.cpp filename changed to config.cpp in
add_library for BUILD_MOD_RADIUS.

HelenMamontova committed:
Radius. The settings.h header file changed to config.h.

HelenMamontova committed:
Radius. The settings.h header file changed to config.h.

HelenMamontova committed:
Radius. The settings.h header file changed to config.h.

HelenMamontova committed:
Radius. The RAD_SETTINGS class name changed to Config.

HelenMamontova committed:
Radius. The RAD_SETTINGS class name changed to Config.

HelenMamontova committed:
Radius. The RAD_SETTINGS class name changed to Config. The m_radSettings
variable class member changed to m_config.

HelenMamontova committed:
Radius. The m_radSettings variable class member changed to m_config.

HelenMamontova committed:
Radius. The RAD_SETTINGS class name changed to Config. The m_radSettings variable class member changed to m_config.
The radSettings parameter name changed to config.

HelenMamontova committed:
Radius. The RAD_SETTINGS class name changed to Config. The m_radSettings variable class member changed to m_config. The radSettings parameter name changed to config in constructor Server.

HelenMamontova committed:
GIT_TAG changed to 1.2.0.

HelenMamontova committed:
GIT_TAG changed to 1.3.0.

HelenMamontova committed:
Header file "rsdproto/attribute.h" added. Server class member variable
m_attribute added.

HelenMamontova committed:
Header file "radproto/attribute_types.h" replaced by "radproto/attribute_codes.h".
The class member m_attribute initialization added to constructor Server.
The attrType variable added. The function type() replaced by code().
The function call make() added to function makeAttributes instead of attributes vector fill code.

HelenMamontova committed:
Header file "radproto/attribute.h" removed. Server class member variable m_attribute removed.

HelenMamontova committed:
Header file "radproto/attribute.h" added. Server class member variable m_attribute initialization removed.
Logic in cycle for is fixed in the makeAttributes function.

madf commented on projects/stargazer/plugins/other/radius/config.cpp:
```diff
@@ -0,0 +1,147 @@
+
+namespace
+{
+    std::string ShowRules(const std::vector<std::pair<std::string, AttrValue>>& attributes)
```
`attributes` -> `rules`

Because these are rules, not attributes.

madf commented on projects/stargazer/plugins/other/radius/config.cpp:
```diff
@@ -0,0 +1,147 @@
+
+std::vector<std::pair<std::string, AttrValue>> Config::ParseRules(const std::string& value, const std::string& paramName)
+{
+    using tokenizer =  boost::tokenizer<boost::char_separator<char>>;
```
Double whitespace after `=`.

madf commented on projects/stargazer/plugins/other/radius/config.cpp:
```diff
@@ -0,0 +1,147 @@
+    for (const auto& token : tokens)
+    {
+        const boost::char_separator<char> sp(" =");
+        const tokenizer tok(token, sp);
```
Consider using `boost::split` here. It gives you `std::vector<std::string>` right away, no need to copy it manually. See the example at the bottom of https://www.boost.org/doc/libs/latest/doc/html/string_algo/usage.html

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -37,27 +44,44 @@ void Server::startReceive()
-    vendorSpecific.push_back(RadProto::VendorSpecific(m_dictionaries.vendorCode("Dlink"), m_dictionaries.vendorAttributeCode("Dlink", "Dlink-User-Level"), vendorValue));
-
-    if (request.type() != RadProto::ACCESS_REQUEST)
+    std::string attrName;
```
`attrName`, `attrValue`, `attrType` and `attrCode` are only used inside the cycle. Please move them there.

HelenMamontova committed:
Parameter name attributes fixed to rules in ShowRules function.

HelenMamontova committed:
Vector keyValue fill replaced with split in the function ParseRules.
Formatting fixed.

HelenMamontova committed:
The variables attrName, attrValue, attrType, attrCode declaration moved
into the cycle in the function makeAttributes.

madf commented on projects/stargazer/plugins/other/radius/config.cpp:
```diff
@@ -2,6 +2,9 @@
 #include <boost/tokenizer.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/algorithm/string/predicate.hpp>
```
Do you need all three? Isn't it enough to include just `string.hpp`, or two: `trim.hpp` and `predicate.hpp`?

madf commented on projects/stargazer/plugins/other/radius/config.cpp:
```diff
@@ -34,20 +37,18 @@ namespace
-        for (const auto& t : tok)
-            keyValue.push_back(t);
+
+        split(keyValue, boost::algorithm::trim_copy_if(token, boost::is_any_of(" \t\r\n\xA0")), boost::is_any_of(" ="), boost::token_compress_on);
```
What is `0xA0`?

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -47,18 +47,20 @@ void Server::startReceive()
         else
             attrValue = at.second.value;
 
+        std::string attrName;
```
```cpp
const auto attrName = at.first;
```
Same for others.

HelenMamontova committed:
The header files <boost/algorithm/string/trim.hpp>, <boost/algorithm/string/predicate.hpp> removed.

HelenMamontova committed:
The variables attrName, attrCode, attrType declaration changed in
function makeAttributes.

HelenMamontova committed:
The predicate for boost::algorithm::trim_copy_if changed in the function
ParseRules.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -124,7 +146,7 @@ bool Server::findUser(const RadProto::Packet& packet)
     {
         m_logger("User's password is incorrect. %s", password.c_str());
```
Please do not expose user password in the log. It's a serious security issue.

madf commented on projects/stargazer/plugins/other/radius/config.cpp:
```diff
@@ -0,0 +1,146 @@
+    {
+        m_errorStr = "Parameter \'Secret\' not found.";
+        printfd(__FILE__, "Parameter 'Secret' not found\n");
+        m_secret = "";
```
I guess, you need to return -1 here.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -37,27 +44,42 @@ void Server::startReceive()
+{
+    const User* user;
+
+    user = findUser(request);
```
Do not look for the user until you really need it. Consider that case when most incoming traffic is garbage, and it won't pass your `request.code() != RadProto::ACCESS_REQUEST` check. You do not need the user at this moment. It would simply burn your CPU for unnecessary lookups. Move it down, to the place where you use it.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
Shouldn't you do something here? E.g. ``startReceive` and return?

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -124,7 +146,7 @@ bool Server::findUser(const RadProto::Packet& packet)
     {
         m_logger("User's password is incorrect. %s", password.c_str());
         printfd(__FILE__, "User's password is incorrect.\n", password.c_str());
```
`password.c_str()` is unused. And shouldn't be. Please remove.

madf commented on projects/stargazer/plugins/other/radius/radius.h:
Please make it private. It is not used anywhere outside the `Run` method.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -1,20 +1,27 @@
+#include <vector>
+#include <string>
+#include <sstream>
+#include <algorithm> // для std::replace
```
First, do not use Cyrillic. Even for comments. Second, you do not use `std::replace`.

madf commented on projects/stargazer/plugins/other/radius/radius.cpp:
```diff
@@ -3,155 +3,16 @@
-
 RADIUS::RADIUS()
     : m_running(false),
       m_users(NULL),
```
Please use `nullptr` instead of `NULL`.

HelenMamontova committed:
The output of the user's password in the log removed.

HelenMamontova committed:
The m_secret variable value changed to -1 if parameter "Secret" not
found in ParseSettings function.

HelenMamontova committed:
The user variable declaration and definition moved in makeResponse
function.

HelenMamontova committed:
The return command added if error in the handleReceive function.

HelenMamontova committed:
The output of the user's password in the function printfd removed.

HelenMamontova committed:
The SetRunning function declaration moved to privat.

HelenMamontova committed:
Header file <algorithm> removed.

HelenMamontova committed:
Null changed to nullptr when initializing m_users in class Radius.

madf commented on projects/stargazer/plugins/other/radius/config.cpp:
```diff
@@ -118,7 +118,7 @@ int Config::ParseSettings(const ModuleSettings & s)
         m_errorStr = "Parameter \'Secret\' not found.";
         printfd(__FILE__, "Parameter 'Secret' not found\n");
-        m_secret = "";
+        m_secret = -1;
```
What? Does it even compile?!

HelenMamontova committed:
The body of the condition if parameter "Secret" not found
fixed and return -1 added in ParseSettings function.
